x-pack/plugins/infra/common/alerting/logs/log_threshold/types.ts

```diff
@@ -301,4 +301,58 @@ export const GroupedSearchQueryResponseRT = rt.intersection([
   }),
 ]);
 
+export type UnoptimizedGroupedSearchQueryResponse = rt.TypeOf<
+  typeof UnoptimizedGroupedSearchQueryResponseRT
+>;
+
+export const OptimizedGroupedSearchQueryResponseRT = rt.intersection([
+  commonSearchSuccessResponseFieldsRT,
+  rt.type({
+    aggregations: rt.type({
+      groups: rt.intersection([
+        rt.type({
+          buckets: rt.array(
+            rt.intersection([
+              rt.type({
+                key: rt.record(rt.string, rt.string),
+                doc_count: rt.number,
+              }),
+              // Chart preview buckets
+              rt.partial({
+                histogramBuckets: rt.type({
+                  buckets: rt.array(chartPreviewHistogramBucket),
+                }),
+              }),
+            ])
+          ),
+        }),
+        rt.partial({
+          after_key: rt.record(rt.string, rt.string),
+        }),
+      ]),
+    }),
+    hits: rt.type({
+      total: rt.type({
+        value: rt.number,
+      }),
+    }),
+  }),
+]);
+
+export type OptimizedGroupedSearchQueryResponse = rt.TypeOf<
+  typeof OptimizedGroupedSearchQueryResponseRT
+>;
+
+export const GroupedSearchQueryResponseRT = rt.union([
+  UnoptimizedGroupedSearchQueryResponseRT,
+  OptimizedGroupedSearchQueryResponseRT,
+]);
+
 export type GroupedSearchQueryResponse = rt.TypeOf<typeof GroupedSearchQueryResponseRT>;
+
+export const isOptimizedGroupedSearchQueryResponse = (
```

Comment:
Theoretically we could have kept one type, and thrown a partial on some properties, but I prefer having the function and type predicate as it makes the separation, and it's reasoning, a bit more explicit imo.